### PR TITLE
Save many queries in products store

### PR DESCRIPTION
**Pull request by vinay-mittal**
Count queries and cached count queries for products are reduced and pagination in content_for :head was not required so it was removed
Now only 3 queries regarding products listing are fired 1 count (pagination) , 1 max query, 1 data load query
Also one count query saved by using length instead of count to load data because it is required on the page.

### CODE REVIEW

**Comment by damianlegawiec** [comment]
:-1: 

Using everywhere `.length` rather than `.count` will only speed things up if caching is disabled. When caching is on (it's on by default on production) it will be slower because you're fetching the whole list (and creating AR objects) rather than only a getting a count needed for cache key.

File: frontend/app/views/spree/products/index.html.erb [hunk 1]
```diff
@@ -12,7 +12,7 @@
 
 <% if params[:keywords] %>
   <div data-hook="search_results">
-    <% if @products.empty? %>
+    <% if !@products.present? %>
```
Comment (priyank-gupta) [inline frontend/app/views/spree/products/index.html.erb]: @vinayvinsol We can use `blank?` here.

File: frontend/app/views/spree/shared/_products.html.erb [hunk 2]
```diff
@@ -1,11 +1,5 @@
-<% content_for :head do %>
-  <% if products.respond_to?(:num_pages) %>
-    <%= rel_next_prev_link_tags products %>
-  <% end %>
-<% end %>
-
 <div data-hook="products_search_results_heading">
-  <% if products.empty? %>
+  <% if !products.present? %>
```
Comment (priyank-gupta) [inline frontend/app/views/spree/shared/_products.html.erb]: @vinayvinsol We can use `blank?` here.

**Comment by jitendra** [comment]
Closing this PR

**Pull request closed without merging**